### ISSUE TITLE
Language about audit is unclear

### ISSUE THREAD

**Issue by fulldecent** [OpenZeppelin/openzeppelin-contracts#2780]
The documentation makes statements about OpenZeppelin Contracts being "audited". These statements are inaccurate at best or they may possibly be called disingenuous.

Here are facts that contributors to this project understand, using plain language, and not possibly accessible to mere *users* of the Contracts project:

1. An audit was performed by a reputable firm of the 2.0.0 product version on October 2018.
2. Nobody should use OpenZeppelin Contracts 2.0.0. At a minimum, this is confirmed by the https://docs.soliditylang.org/en/latest/, and I hope it is not necessary for me to provide further detail here.
3. The audit, the OpenZeppelin Contracts project, and the OpenZeppelin company all confirm that as soon as you touch anything, the audit becomes inapplicable.
4. The audits that were performed in the past are only useful to historians and researchers. They have zero attestation of quality or safety on the current state of the project. (Which of course, has been touched since 2.0.0.)

So the following changes should be made.

1. Old audits are dangerous and should be removed from repository. They are historical, and can live permanently in the git history.
2. If for marketing reasons it is desirable to mention these past audits, they must be discussed in the historical context.
3. Never mention "full audit". This implies there is some other kind of audit, which there isn't.

Other notes are outside the scope of this issue:

- Design guideline D6 says "We will perform regular code audits on major releases, and hire security professionals to provide independent review"
- Readme / security says "OpenZeppelin is meant to provide tested and community-audited code"
- The last audit was in 2018 when Bitcoin cost USD 6,000

These three statements seem out of synchronization.

**Comment by frangio** [comment]
> Nobody should use OpenZeppelin Contracts 2.0.0. At a minimum, this is confirmed by the https://docs.soliditylang.org/en/latest/, and I hope it is not necessary for me to provide further detail here.

As we've discussed elsewhere, this is not exactly the version policy that we have, and the Solidity documentation is not particularly relevant (see https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2759#issuecomment-891387641). I've now made the version policy explicit, in particular in [SECURITY.md](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/SECURITY.md), where it can be seen that OpenZeppelin Contracts 2.x is indeed supported for critical bug fixes. The exact version 2.0.0 should likely not be used, in favor of the latest 2.x release.

The extent to which the 2.0.0 audit applies to the latest 2.x release (or even 3.x and 4.x for that matter) is a question of subjective risk assessment that we can't make for others, but this does not imply that we should bury past audits in git history. An audit is in fact never an attestation of quality or safety as you imply (I'm confident any auditor will agree with me on that), but an analysis that continues to have value even after changes are made.

These are relevant documents that clearly state they were made for a particular past commit and in a past date. I've removed a potentially misleading comment in https://github.com/OpenZeppelin/openzeppelin-contracts/commit/56b3b7a4879fee4addcf77f70d519b7c8635309f and I believe the language is clear now.
